presto-iceberg/src/main/java/com/facebook/presto/iceberg/RefsTable.java

```diff
@@ -0,0 +1,102 @@
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.iceberg.util.PageListBuilder;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.FixedPageSource;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.SystemTable;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.Table;
+
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class RefsTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = ImmutableList.<ColumnMetadata>builder()
+            .add(new ColumnMetadata("name", VARCHAR))
+            .add(new ColumnMetadata("type", VARCHAR))
+            .add(new ColumnMetadata("snapshot_id", BIGINT))
+            .add(new ColumnMetadata("max_reference_age_in_ms", BIGINT))
+            .add(new ColumnMetadata("min_snapshots_to_keep", BIGINT))
+            .add(new ColumnMetadata("max_snapshot_age_in_ms", BIGINT))
+            .build();
+
+    public RefsTable(SchemaTableName tableName, Table icebergTable)
+    {
+        tableMetadata = new ConnectorTableMetadata(requireNonNull(tableName, "tableName is null"), COLUMNS);
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages(tableMetadata, icebergTable));
+    }
+
+    private static void appendValue(Long value, PageListBuilder pagesBuilder)
+    {
+        if (value == null) {
+            pagesBuilder.appendNull();
+        }
+        else {
+            pagesBuilder.appendBigint(value);
+        }
+    }
+
+    private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table icebergTable)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
+
+        icebergTable.refs().forEach((key, value) -> {
+            pagesBuilder.beginRow();
+            pagesBuilder.appendVarchar(key);
+            pagesBuilder.appendVarchar(value.isTag() ? "TAG" : "BRANCH");
```

Comment:
What do you think about doing `String.valueOf(value.type())`? This way it handles updates to the `SnapshotRefType` enum.